src/parser/core/modules/NormalisedEvents.ts

```diff
@@ -0,0 +1,178 @@
+import {DamageEvent, Event, HealEvent} from 'fflogs'
+import Module, {dependency} from 'parser/core/Module'
+import HitType from 'parser/core/modules/HitType'
+
+// Based on multi-hit margin previously in use for barrage and AOE modules
+const LEGACY_MUTLIHIT_DEDUPLICATION_TIME_WINDOW = 500
+
+const DAMAGE_EVENT_TYPES: Array<Event['type']> = [
+	'damage',
+	'calculateddamage',
+]
+
+const HEAL_EVENT_TYPES: Array<Event['type']> = [
+	'heal',
+	'calculatedheal',
+]
+
+export interface NormalisedDamageEvent extends Omit<DamageEvent, 'type'> {
```

Comment:
General _thoughts_ on the attrs themselves:

* Hm... I'm aware that "ghost" is a popular term for the issue we're modelling, but it's also a term that's not very self-explanatory, and often has to be explained to people when it gets brought up. I'm not opposed to referencing it as ghosting in xiva, but there might be value in putting some thought into it to be sure it's what we want to codify. If it is, I'll probably req. a comment explaining what it means.
* Events don't _need_ to be flat structures. I'm not coming up with anything clever right now, but I wonder if there's a neater representation with a nested structure? Maybe just something to consider.